public/app/features/dashboard-scene/serialization/transformSaveModelSchemaV2ToScene.ts

```diff
@@ -566,3 +571,19 @@ export function getPanelElement(dashboard: DashboardV2Spec, elementName: string)
 export function getLibraryPanelElement(dashboard: DashboardV2Spec, elementName: string): LibraryPanelKind | undefined {
   return dashboard.elements[elementName].kind === 'LibraryPanel' ? dashboard.elements[elementName] : undefined;
 }
+
+function createElementPanelMapping(dashboard: DashboardV2Spec) {
```

Comment:
I guess the singleton exists to have it here, in this transformer 🤔 

What if we create it as this function does, but it is returned to the transformer, and before returning the created DashboardScene, we do something like `scene.setPersistingElementUID(mapping)`?